lib/IRGen/GenCall.cpp

```diff
@@ -2587,8 +2588,10 @@ llvm::Value *emitIndirectCoroFunctionPointer(IRGenFunction &IGF,
       IntPtrTy, APInt(IntPtrTy->getBitWidth(), 0));
   llvm::Constant *One = llvm::Constant::getIntegerValue(
       IntPtrTy, APInt(IntPtrTy->getBitWidth(), 1));
+  // TODO: Avoid implicit truncation.
   llvm::Constant *NegativeOne = llvm::Constant::getIntegerValue(
-      IntPtrTy, APInt(IntPtrTy->getBitWidth(), -2));
+      IntPtrTy, APInt(IntPtrTy->getBitWidth(), -2, /*isSigned*/ false,
+                      /*implicitTrunc*/ true));
```

Comment:
You should pass `/*isSigned*/ true`. If `IntPtrTy->getBitWidth()` were actually huge, we would in fact want this to be sign-extended, and it also fixes the assertion because it's okay to truncate consistent sign bits.

Reply:
Thanks! 

> how did we end up with a variable called NegativeOne that represents a value of -2

https://github.com/swiftlang/swift/commit/68bc33fed32#diff-e3cb1b9416734345839b90f1c5ec26d9173d335e687abb007c9ec0b332b3d3e4

cc @compnerd